src/plugins/dev_tools/public/application.tsx

```diff
@@ -154,51 +154,40 @@ function DevToolsWrapper({
-              {devTools.map((currentDevTool) => (
-                <EuiToolTip content={currentDevTool.tooltipContent} key={currentDevTool.id}>
-                  <EuiTab
-                    disabled={currentDevTool.isDisabled()}
-                    isSelected={currentDevTool === activeDevTool}
-                    onClick={() => {
-                      if (!currentDevTool.isDisabled()) {
-                        updateRoute(`/${currentDevTool.id}`);
-                      }
-                    }}
-                  >
-                    {currentDevTool.title}
-                  </EuiTab>
-                </EuiToolTip>
-              ))}
-            </EuiTabs>
-          </EuiFlexItem>
-          {dataSourceEnabled ? (
-            <EuiFlexItem grow={false} className="dataSourceSelector">
-              <EuiComboBox
-                aria-label={i18n.translate('devTool.devToolWrapper.DataSourceComboBoxAriaLabel', {
-                  defaultMessage: 'Select a Data Source',
-                })}
-                placeholder={i18n.translate(
-                  'devTool.devToolWrapper.DataSourceComboBoxPlaceholder',
-                  {
-                    defaultMessage: 'Select a Data Source',
-                  }
-                )}
-                singleSelection={{ asPlainText: true }}
-                options={dataSources}
-                selectedOptions={selectedOptions}
-                onChange={onChange}
-                prepend="DataSource"
-                compressed
-                isDisabled={!dataSourceEnabled}
-              />
-            </EuiFlexItem>
-          ) : null}
-        </EuiFlexGroup>
-      </div>
+      <EuiTabs className="devAppTabs">
+        {devTools.map((currentDevTool) => (
+          <EuiToolTip content={currentDevTool.tooltipContent} key={currentDevTool.id}>
+            <EuiTab
+              disabled={currentDevTool.isDisabled()}
+              isSelected={currentDevTool === activeDevTool}
+              onClick={() => {
+                if (!currentDevTool.isDisabled()) {
+                  updateRoute(`/${currentDevTool.id}`);
+                }
+              }}
+            >
+              {currentDevTool.title}
+            </EuiTab>
+          </EuiToolTip>
+        ))}
+        <div className="devAppDataSourcePicker">
+          <EuiComboBox
+            aria-label={i18n.translate('devTool.devToolWrapper.DataSourceComboBoxAriaLabel', {
+              defaultMessage: 'Select a Data Source',
+            })}
+            placeholder={i18n.translate('devTool.devToolWrapper.DataSourceComboBoxPlaceholder', {
+              defaultMessage: 'Select a Data Source',
+            })}
+            singleSelection={{ asPlainText: true }}
+            options={dataSources}
+            selectedOptions={selectedOptions}
+            onChange={onChange}
+            prepend="DataSource"
+            compressed
+            isDisabled={!dataSourceEnabled}
+          />
+        </div>
```

Comment:
So one major change here is now that this data source picker will always be visible, regardless of whether the feature is enabled or disabled. On the one hand, it helps a bit with discoverability of MD, but on the other hand it's a waste of space and adds visual noise for users who have purposely chosen to disable. @KrooshalUX is there general design system guidance for this type of config-disabled element (as opposed to an element that may or may not be disabled based on some sort of application state)?

Reply:
good catch, it's actually I forget to add the conditional render when I refactor this part of the code. I'll add it back, so if user disable datasource feature, datasource picker won't show, user will get the same old dev tool console experience.